### PR TITLE
Fix indexation of CJK languages.

### LINKED ISSUE #802: Support CJK index creation and query
Current situation:
https://download.kiwix.org/zim/wikipedia/wikipedia_zh_mathematics_mini_2023-05.zim

fulltext search `平方`  
gives no result.
image

actually there are much more articles contain this word.
image
image

english word seems ok.
image
The `No` seems another issue.

CJK characters have to be indexed with CJK flags

Creation:
```
   Xapian::TermGenerator indexer;

    indexer.set_flags( Xapian::TermGenerator::FLAG_CJK_NGRAM );
```
Query:
```
      Xapian::QueryParser::feature_flag flag = Xapian::QueryParser::FLAG_DEFAULT;
      if( searchMode == FTS::Wildcards )
        flag = Xapian::QueryParser::FLAG_WILDCARD;
      Xapian::Query query = qp.parse_query( query_string, flag|Xapian::QueryParser::FLAG_CJK_NGRAM );
```

_Originally posted by @kelson42 in https://github.com/openzim/libzim/issues/794#issuecomment-1573246698_

### PR DESCRIPTION
**Pull request by mgautierfr**
Fix #802

### CODE REVIEW

**Comment by kelson42** [comment]
@mgautierfr Can you please provide an index size benchmark?

File: test/suggestion.cpp [hunk 1]
```diff
@@ -667,4 +667,39 @@ TEST(Suggestion, reuseSearcher) {
   ASSERT_EQ(count, 3);
 }
 
+TEST(Suggestion, CZK) {
```
Comment (veloman-yunkan) [inline test/suggestion.cpp:670]: CZK -> CJK

File: test/search.cpp [hunk 2]
```diff
@@ -245,4 +245,45 @@ TEST(Search, geoQuery)
...
+  searcher.addArchive(archive);
+  searcher.setVerbose(true);
+
+  {
+    zim::Query query("平方");
+    auto search = searcher.search(query);
+
+    // CJK language is "special". Words are not separated by spaces.
+    // We are searching for corresponding "combinations" of ideograms.
+    // So `平方` must also found `平方根`
```
Comment (veloman-yunkan) [inline test/search.cpp:274]: found -> find

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]
Is the commit containing the fix final? If yes, why does it have [WIP] in its title?

**Comment by mgautierfr** [comment]
> Also, do you have an explanation for why the codecov/patch flow has failed?

For a unknown reason (maybe the `assert` at the beginning of function ?), codecov thinks the line are partially tested.
Maybe it is not totally counted as tested and, as we add new lines, the coverage decrease.

**Review by veloman-yunkan — APPROVED** [review]
Don't merge without squashing the fixup commit.
*Don't merge without squashing the fixup commit.*
**Don't merge without squashing the fixup commit.**
***Don't merge without squashing the fixup commit.***

**Pull request merged**